include/aspect/utilities.h

```diff
@@ -411,6 +427,44 @@ namespace aspect
       return std::vector<T> ();
     }
 
+    template <int dim>
+    std::vector<std::string>
+    expand_dimensional_variable_names (const std::vector<std::string> &var_declarations)
+    {
+      std::vector<std::string> dim_names = {"x", "y", "z"};
+      char fn_split = "(", fn_end = ")";
+      std::vector<std::string> var_name_list;
+      for (auto const var_decl : var_decls)
+        {
+
+          if (var_decl.find(fn_split) != std::string::npos&&var_decl[var_decl.length()-1]==fn_end)
+            {
+              std::string fn_name = var_decl.substr(0, var_decl.find(fn_split));
+              std::string var_name = var_decl.substr(var_decl.find(fn_split), var_decl.length()-2-var_decl.find(fn_split));
```

Comment:
have you tried what happens if someone writes `tensor ( u ) `? I think it would be useful to each whitespaces around parentheses.